app/services/listing_service/api/categories.rb

```diff
@@ -2,12 +2,24 @@ module ListingService::API
   CategoryStore = ListingService::Store::Category
 
   class Categories
-    def get(community_id:)
+    def get_all(community_id:)
       Result::Success.new(
         CategoryStore.get_all(community_id: community_id)
       )
     end
 
+    def get(community_id:, category_id:)
+      res = HashUtils.deep_find(CategoryStore.get_all(community_id: community_id), :children) { |cat|
```

Comment:
The deep_find is a neat helper function but shouldn't this be a direct lookup operation to DB? If the store would just directly give you the object then we wouldn't need a tree traversal here, right?